### PR TITLE
Cart UI

### PR DESCRIPTION
**Pull request by abhijitdotsharma**
Cart Layout and Cart Functionality added
- Add to Cart
- Remove from cart
- Clear Cart
- Cart Summary

Important files to check 
- `src/pages/cart-page`
- `src/components/cart`

Not to check 
- CSS
- Layout responsiveness

Styling to be improved on future PR

[View Deploy Preview](https://deploy-preview-5--delightful-elf-7675a4.netlify.app/)

### CODE REVIEW

File: src/components/cart/CartProduct.jsx [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    const { addToCart, removeFromCart, decreaseProductCountInCart } = useCart();
+
+    function getDiscountOfProduct(price, originalPrice) {
+        //discPercent = originalPrice - sellingPrice / originalPrice * 100
+        return parseInt(((Number(originalPrice) - Number(price)) / Number(originalPrice)) * 100)
+    }
+
+    let discount = getDiscountOfProduct(product.price, product.originalPrice)
+
+    return (
```
Comment (mariyasada) [inline src/components/cart/CartProduct.jsx:14]: you can destructure the product at the above like {price,originalPrice,photo,title}=product

File: src/components/cart/CartProducts.jsx [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import CartProduct from "./CartProduct"
+export const CartProducts = ({ cart }) => {
+    return (
+        <div style={{ position: 'relative', top: '4rem', left: '-3rem' }}>
+            {cart.cartItems.map(product => <CartProduct key={product.id} product={product} />)}
+            {/* The map here will result in unique product only because, if same product added, it's count
```
Comment (mariyasada) [inline src/components/cart/CartProducts.jsx:6]: avoid inline styling

File: src/components/cart/cart-product.css [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+    color: var(--white);
+    width: 70%;
+    padding: 0.5rem 2rem;
+    margin-top: 1rem;
+}
+/*Turn off number input spinner*/
+input[type=number]::-webkit-inner-spin-button, 
+input[type=number]::-webkit-outer-spin-button { 
+  -webkit-appearance: none; 
+  margin: 0;
```
Comment (mariyasada) [inline src/components/cart/cart-product.css:89]: good to see this type css

File: src/components/products/ProductCard.jsx [hunk 4]
```diff
@@ -1,6 +1,15 @@
 //I will take a product detail prop from Products.jsx and render it
+import { useCart } from "../../context/cart-context";
 
 const ProductCard = ({ product }) => {
+    const { addToCart } = useCart();
+
+    function getDiscountOfProduct(price, originalPrice) {
+        //discPercent = originalPrice - sellingPrice / originalPrice * 100
+        return parseInt(((Number(originalPrice) - Number(price)) / Number(originalPrice)) * 100)
```
Comment (mariyasada) [inline src/components/products/ProductCard.jsx:9]: here also you can destructure product

**Review by mariyasada — APPROVED** [review]
added some suggestions, check why the preview is not working. when I click to check it out it redirects to some HTML page, check all the links

File: .env [hunk 5]
```diff
@@ -0,0 +1 @@
+REACT_APP_JWT_SECRET=NERIST
```
Comment (harshalini) [inline .env:1]: Don't push .env file directly

**Review by harshalini — APPROVED** [review]
Overall code is nice and readable

**Pull request merged**